File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;

    const arr = s.split("")
    let cnt = 0
    
    for (let i = 0; i < arr.length; i++){
        if (arr[i] == '('){
            cnt ++;
        }
        else{
            cnt --;
        }
        
        if (cnt < 0){
            answer = false;
        }
    }
    
    if (cnt !== 0){
        answer = false;
    }

    return answer;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Deque<String> stack = new ArrayDeque<>();
        
        for (int i = 0; i < s.length(); i++){
            String text = s.substring(i,i+1);
            
            if (text.equals("(")){
                stack.push(text);
            }
            else{
                if (stack.isEmpty()){
                    return false;
                }
                stack.pop();
            }
        }
        
        if (!stack.isEmpty()){
            answer = false;
        }
        

        return answer;
    }
}